### PR TITLE
fix: do not load cache files when recompile flag is set

### PR DESCRIPTION
**Pull request by caijw**
Fix issue https://github.com/denoland/deno/issues/1637 which is labeled bug.
if recompile flag is set, not load cache files,
else load cache files.
And add a test case test_code_fetch_1.

### CODE REVIEW

**Comment by kitsonk** [comment]
LGTM, feels like a good clean solution, but I am not the best person for having an opinion on Rust code.

File: src/deno_dir.rs [hunk 1]
```diff
@@ -313,31 +319,41 @@ impl DenoDir {
...
+      media_type: out.media_type,
+      source_code: out.source_code,
+      maybe_output_code_filename: output_code_filename
+        .to_str()
+        .map(|s| s.to_string()),
+      maybe_output_code: maybe_output_code,
+      maybe_source_map_filename: output_source_map_filename
+        .to_str()
+        .map(|s| s.to_string()),
+      maybe_source_map: maybe_source_map,
```
Comment (kevinkassimo) [inline src/deno_dir.rs]: nit: could be simplified as 
```rust
Ok(CodeFetchOutput {
  // ...
  maybe_source_code,
  maybe_source_map,
})
```

File: src/deno_dir.rs [hunk 2]
```diff
@@ -313,31 +319,41 @@ impl DenoDir {
...
+      filename: out.filename,
+      media_type: out.media_type,
+      source_code: out.source_code,
+      maybe_output_code_filename: output_code_filename
+        .to_str()
+        .map(|s| s.to_string()),
+      maybe_output_code: maybe_output_code,
+      maybe_source_map_filename: output_source_map_filename
+        .to_str()
+        .map(|s| s.to_string()),
```
Comment (ry) [inline src/deno_dir.rs]: nit: could be simplified as `map(String::from)` (here and elsewhere)

File: src/deno_dir.rs (hunk 1 above)
Comment (caijw) [reply src/deno_dir.rs]: done ~

File: src/deno_dir.rs (hunk 2 above)
Comment (caijw) [reply src/deno_dir.rs]: done ~

**Pull request merged**